src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysis.java

```diff
@@ -328,10 +332,26 @@ private PostContingencyResult runPostContingencySimulation(LfNetwork network, Ac
         return new PostContingencyResult(lfContingency.getContingency(), postContingencyComputationOk, postContingencyLimitViolations);
     }
 
-    private void deactivateEquations(LfContingency lfContingency, EquationSystem equationSystem, List<Equation> deactivatedEquations, List<EquationTerm> deactivatedEquationTerms) {
+    private void deactivateEquations(LfContingency lfContingency, EquationSystem equationSystem, VariableSet variableSet,
+                                     List<Equation> deactivatedEquations, List<EquationTerm> deactivatedEquationTerms) {
         for (LfBranch branch : lfContingency.getBranches()) {
             LOGGER.trace("Remove equations and equations terms related to branch '{}'", branch.getId());
 
+            // deactivate equation of voltage discrete controlled bus
+            if (branch.isVoltageController()) {
+                DiscreteVoltageControl dvc = branch.getDiscreteVoltageControl();
+                boolean sharedControl = dvc.getControllers().size() > 1;
+                if (sharedControl) {
+                    removeSharedVoltageControlEquation(branch, equationSystem, variableSet, deactivatedEquations);
+                    dvc.removeController(branch);
```

Comment:
I understand that you need to modify the discrete voltage control but removing the controller lost during the contingency. I think that you made that for the transformer voltage control outerloop. But, for sure you have to restore it after the post contingency simulation. Maybe the solution is to add a boolean list in the discrete voltage control to say if a controller is active or not. Note that we will have issues only if transformer voltage control parameter is active.